apps/minifront/src/components/swap/swap-form/simulate-swap.tsx

```diff
@@ -3,16 +3,33 @@ import { SimulateSwapResult } from './simulate-swap-result';
 import { AllSlices } from '../../../state';
 import { useStoreShallow } from '../../../utils/use-store-shallow';
 import { EstimateButton } from './estimate-button';
+import { useEffect } from 'react';
 
 const simulateSwapSelector = (state: AllSlices) => ({
   simulateSwap: state.swap.instantSwap.simulateSwap,
+  resetSimulateSwap: state.swap.instantSwap.reset,
   disabled:
     state.swap.txInProgress || !state.swap.amount || state.swap.instantSwap.simulateSwapLoading,
   result: state.swap.instantSwap.simulateSwapResult,
+  amount: state.swap.amount,
 });
 
+// Automatically trigger swap simulation on user input
 export const SimulateSwap = ({ layoutId }: { layoutId: string }) => {
-  const { simulateSwap, disabled, result } = useStoreShallow(simulateSwapSelector);
+  const { simulateSwap, resetSimulateSwap, disabled, result, amount } =
+    useStoreShallow(simulateSwapSelector);
+
+  useEffect(() => {
```

Comment:
Not critical to fix, but in the future, we should always see if it's possible to use react-query first for all async tasks. The complexities of loading/error/cache invalidation is abstracted away.

Reply:
totally, sorry about that – that's an easy refactor